src/plugins/discover/public/application/main/services/discover_state.ts

```diff
@@ -382,6 +383,23 @@ export function getDiscoverStateContainer({
       fetchData();
     });
 
+    services.data.search.session.enableStorage(
+      createSearchSessionRestorationDataProvider({
+        appStateContainer,
+        data: services.data,
+        getSavedSearch: () => savedSearchContainer.getState(),
```

Comment:
Wonder if this logic should be executed again when `savedSearchContainer` state changes.

Reply:
It's just using `id` and `name` of a saved search when it's needed to name a search session ... 

https://github.com/elastic/kibana/blob/d585de9fefc05ca9cf689928bdd9ddfd29ce502a/src/plugins/discover/public/application/main/services/discover_state.ts#L459

So it should be fine, but I bet this code a bit in the future :)